clisops/utils/dataset_utils.py

```diff
@@ -368,6 +368,556 @@ def adjust_date_to_calendar(da, date, direction="backwards"):
     )
 
 
+def add_hor_CF_coord_attrs(
+    ds, lat="lat", lon="lon", lat_bnds="lat_bnds", lon_bnds="lon_bnds", keep_attrs=False
+):
+    """
+    Add the common CF variable attributes to the horizontal coordinate variables.
+
+    Parameters
+    ----------
+    lat : str, optional
+        Latitude coordinate variable name. The default is "lat".
+    lon : str, optional
+        Longitude coordinate variable name. The default is "lon".
+    lat_bnds : str, optional
+        Latitude bounds coordinate variable name. The default is "lat_bnds".
+    lon_bnds : str, optional
+        Longitude bounds coordinate variable name. The default is "lon_bnds".
+    keep_attrs : bool, optional
+        Whether to keep original coordinate variable attributes if they do not conflict.
+        In case of a conflict, the attribute value will be overwritten independent of this setting.
+        The default is False.
+
+    Returns
+    -------
+    xarray.Dataset
+        The input dataset with updated coordinate variable attributes.
+    """
+    # Define common CF coordinate variable attrs
+    lat_attrs = {
+        "bounds": "lat_bnds",
+        "units": "degrees_north",
+        "long_name": "latitude",
+        "standard_name": "latitude",
+        "axis": "Y",
+    }
+    lon_attrs = {
+        "bounds": "lon_bnds",
+        "units": "degrees_east",
+        "long_name": "longitude",
+        "standard_name": "longitude",
+        "axis": "X",
+    }
+
+    # Overwrite or update coordinate variables of input dataset
+    try:
+        if keep_attrs:
+            ds[lat].attrs.update(lat_attrs)
+            ds[lon].attrs.update(lon_attrs)
+        else:
+            ds[lat].attrs = lat_attrs
+            ds[lon].attrs = lon_attrs
+    except KeyError:
+        raise KeyError("Not all specified coordinate variables exist in the dataset.")
+
+    return ds
+
+
+def reformat_SCRIP_to_CF(ds, keep_attrs=False):
+    """
+    Reformat dataset from SCRIP to CF format.
```

Comment:
`ESMF.Grid` objects can be created from SCRIP files, so I'm wondering if everything here is really needed ? 

xESMF has a `ds_to_ESMFGrid` that could maybe work here ?